### PR TITLE
[17.0][ADD] product_document_domain

### PR DESCRIPTION
**Pull request by Tisho99**
This module edits the product document smartbutton domain to show the
documents of the current product record, and the documents of the parent
(or template) if exists. The default behaviour is the oposite.

T-6750

### CODE REVIEW

**Comment by Tisho99** [comment]
@ValentinVinagre @HaraldPanten

**Review by sara-castello — APPROVED** [review]
functional review LGTM

File: product_document_domain/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def action_open_documents(self):
+        res = super().action_open_documents()
+        res["context"].update(
+            {
+                "default_res_model": self._name,
```
Comment (manuelregidor) [inline product_document_domain/models/product_product.py]: super method already returns "default_res_model": self._name in res["context"].

File: product_document_domain/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                "default_res_id": self.id,
```
Comment (manuelregidor) [inline product_document_domain/models/product_product.py]: super method already returns "default_res_id": self.id in res["context"].

File: product_document_domain/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                "default_parent_res_id": self.product_tmpl_id.id,
+                "search_default_context_variant": False,
+                "search_default_context_template": False,
+                "search_default_context_variant_template": True,
+            }
+        )
+        return res
+
+    def _compute_product_document_count(self):
```
Comment (manuelregidor) [inline product_document_domain/models/product_product.py]: Have you thought of calling super and then adding the number of records that meet the domain  ("res_model", "=", "product.template"), ("res_id", "=", template.id)?

File: product_document_domain/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _compute_product_document_count(self):
```
Comment (manuelregidor) [inline product_document_domain/models/product_template.py:10]: Kind of similar here. You could call super method and then reduce the records with domain ('res_model', '=', 'product.product'), ('res_id', 'in', template.product_variant_ids.ids). Has it been completely completely overriden for any specific reason?

**Review by manuelregidor — CHANGES_REQUESTED** [review]

**Comment by Tisho99** [comment]
@manuelregidor can you review again?

**Review by manuelregidor — APPROVED** [review]
Technical review. LGTM

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by ValentinVinagre** [comment]
@HaraldPanten could u review please?

**Review by HaraldPanten — APPROVED** [review]
LGTM

**Pull request merged**